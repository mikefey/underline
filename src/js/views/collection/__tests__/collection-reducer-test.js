import test from 'ava';
import {
  DB_READY,
} from './../../../libs/indexeddb/indexeddb-constants';
import {
  DOWNLOAD_BOOK,
  DOWNLOAD_BOOK_COMPLETE,
  GET_COLLECTION,
  GET_COLLECTION_COMPLETE,
  ADD_BOOK_TO_COLLECTION,
  UPDATE_COLLECTION,
} from './../collection-constants';
import collectionReducer from './../collection-reducer.js';

test('default state', (t) => {
  t.snapshot(collectionReducer(undefined, {}));
});

test('handles DB_READY', (t) => {
  t.snapshot(collectionReducer(undefined, { type: DB_READY }));
});

test('handles DOWNLOAD_BOOK', (t) => {
  t.snapshot(collectionReducer(undefined, { type: DOWNLOAD_BOOK }));
});

test('handles DOWNLOAD_BOOK_COMPLETE', (t) => {
  t.snapshot(collectionReducer(undefined, { type: DOWNLOAD_BOOK_COMPLETE }));
});

test('handles GET_COLLECTION', (t) => {
  t.snapshot(collectionReducer(undefined, { type: GET_COLLECTION }));
});

test('handles GET_COLLECTION_COMPLETE', (t) => {
  t.snapshot(collectionReducer(undefined, { type: GET_COLLECTION_COMPLETE }));
});

test('handles UPDATE_COLLECTION', (t) => {
  t.snapshot(collectionReducer(undefined, {
    type: UPDATE_COLLECTION,
    data: [{ id: '1', title: 'Book 1' }, { id: '2', title: 'Book 2' }],
  }));
});

test('handles ADD_BOOK_TO_COLLECTION', (t) => {
  t.snapshot(collectionReducer(undefined, {
    type: ADD_BOOK_TO_COLLECTION,
    data: [{ id: '1', title: 'Book 1' }, { id: '2', title: 'Book 2' }],
  }));
});
