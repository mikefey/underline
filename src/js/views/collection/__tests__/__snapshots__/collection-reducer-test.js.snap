// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default state 1`] = `
Object {
  "collection": Array [],
  "dbReady": false,
  "downloadingBook": false,
  "gettingCollection": false,
  "initialCollectionLoaded": false,
}
`;

exports[`handles ADD_BOOK_TO_COLLECTION 1`] = `
Object {
  "collection": Array [
    Object {
      "id": "1",
      "title": "Book 1",
    },
    Object {
      "id": "2",
      "title": "Book 2",
    },
  ],
  "dbReady": false,
  "downloadingBook": false,
  "gettingCollection": false,
  "initialCollectionLoaded": false,
}
`;

exports[`handles DB_READY 1`] = `
Object {
  "collection": Array [],
  "dbReady": true,
  "downloadingBook": false,
  "gettingCollection": false,
  "initialCollectionLoaded": false,
}
`;

exports[`handles DOWNLOAD_BOOK 1`] = `
Object {
  "collection": Array [],
  "dbReady": false,
  "downloadingBook": true,
  "gettingCollection": false,
  "initialCollectionLoaded": false,
}
`;

exports[`handles DOWNLOAD_BOOK_COMPLETE 1`] = `
Object {
  "collection": Array [],
  "dbReady": false,
  "downloadingBook": false,
  "gettingCollection": false,
  "initialCollectionLoaded": false,
}
`;

exports[`handles GET_COLLECTION 1`] = `
Object {
  "collection": Array [],
  "dbReady": false,
  "downloadingBook": false,
  "gettingCollection": true,
  "initialCollectionLoaded": false,
}
`;

exports[`handles GET_COLLECTION_COMPLETE 1`] = `
Object {
  "collection": Array [],
  "dbReady": false,
  "downloadingBook": false,
  "gettingCollection": false,
  "initialCollectionLoaded": false,
}
`;

exports[`handles UPDATE_COLLECTION 1`] = `
Object {
  "collection": Array [
    Object {
      "id": "1",
      "title": "Book 1",
    },
    Object {
      "id": "2",
      "title": "Book 2",
    },
  ],
  "dbReady": false,
  "downloadingBook": false,
  "gettingCollection": false,
  "initialCollectionLoaded": true,
}
`;
